modules/local/fastp/main.nf

```diff
@@ -0,0 +1,51 @@
+process FASTP {
+    tag "${meta.id}"
+    label 'process_medium'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/fastp:0.23.4--hadf994f_2' :
+        'biocontainers/fastp:0.23.4--hadf994f_2' }"
+
+    input:
+    tuple val(meta), path(reads_fwd), path(reads_rev)
+    val max_fastq_records
+
+    output:
+    tuple val(meta), path('*_R1.fastp.fastq.gz'), path('*_R2.fastp.fastq.gz'), emit: fastq
+    path 'versions.yml'                                                      , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+
+    """
+    fastp \\
+        ${args} \\
+        --in1 ${reads_fwd} \\
+        --in2 ${reads_rev} \\
+        --disable_quality_filtering \\
+        --disable_length_filtering \\
+        --disable_adapter_trimming \\
+        --disable_trim_poly_g \\
+        --split_by_lines ${4 * max_fastq_records} \\
+        --thread ${task.cpus} \\
+        --out1 ${meta.sample_id}_${meta.library_id}_${meta.lane}_R1.fastp.fastq.gz \\
+        --out2 ${meta.sample_id}_${meta.library_id}_${meta.lane}_R2.fastp.fastq.gz
```

Comment:
just checking that you are not intending to publishing or passing the logs out for qc reporting?

Reply:
Yeah that's correct, there is no intention to publish QC or logs for reporting in the first release - the idea was to keep upstream alignment very simple for the first iteration. If the core team is onboard then I'll plan to implement more QC and integrate with MultiQC after the first release!

Reply:
Created an issue to consider this as an extension in after the initial release:

- https://github.com/nf-core/oncoanalyser/issues/83